### PR TITLE
fix(chart): keep point labels inside the plot bounding box / 图表点标签严格限制在绘图区域内

### PR DESCRIPTION
**Pull request by cquil11**
## Summary

Point labels could escape the plot area — a point near the top edge kept its default above-the-point slot and the zoom-group clip path sliced the label in half (e.g. `5xDEP8+1xDEP16` on the Throughput per Utility MW chart). The plot clip rect is now a **strict bounding box** for every point label.

## What changed

- `lib/d3-chart/plot-bounds.ts`: new `plotClipSize(zoomGroup)` reads the `<clipPath>` rect the zoom group is clipped to (in zoom-group units); `plotBounds()` now shares it.
- `inference/ui/line-label-layer.ts`: `placePointLabels()` is the shared layout pass. Each label tries a ladder of slots (above → below → further above → further below) and only accepts one that fits entirely inside the plot, sliding horizontally at the left/right edge instead of spilling past it. A label with no eligible slot is hidden rather than drawn half-clipped. `avoidPointLabelCollisions()` (ScatterGraph, with collision avoidance) and `keepPointLabelsInPlot()` (bounding box only) are thin wrappers. Charts that don't clip (`clipContent: false`) get no constraint.
- `inference/ui/GPUGraph.tsx`: the compare chart had no placement pass at all; it now runs `keepPointLabelsInPlot` on render, zoom, and when the label toggle turns labels back on (via `onDisplayUpdate`).

## Unofficial-run overlays

Overlay points (`?unofficialrun=`) render through the same `.dot-group .point-label` DOM in the same clipped zoom group, so the bounding-box pass covers them identically — no separate code path.

## Tests

- New `point-label-placement.test.ts` (jsdom): reads the clip rect, flips below at the top edge, climbs above at the bottom edge, slides inward at the left/right edges, hides when nothing fits or the point is zoomed out of the plot, resets prior shifts, no constraint when unclipped, explicit bounds, and `keepPointLabelsInPlot` skipping collisions.
- `line-label-layer.test.ts`: primitives `horizontalShiftIntoBounds` / `fitsVertically`.
- `plot-bounds.test.ts`: `plotClipSize` incl. missing/degenerate clip and detached group.
- `bun run typecheck`, `lint`, `fmt`, `check:typography`, and the inference + d3-chart vitest suites pass locally.

## 中文说明

点标签有时会超出绘图区域：靠近顶部边缘的点保留默认的上方标签位置，zoom-group 的裁剪路径把标签切掉一半（例如 Throughput per Utility MW 图中的 `5xDEP8+1xDEP16`）。现在将绘图区的裁剪矩形作为所有点标签的**严格包围盒**。

- `plot-bounds.ts` 新增 `plotClipSize()`，读取 zoom group 所引用的 `<clipPath>` 矩形尺寸；`plotBounds()` 复用它。
- `line-label-layer.ts` 新增共享布局函数 `placePointLabels()`：每个标签依次尝试上方 / 下方 / 更上方 / 更下方，只接受完全落在绘图区内的位置，靠近左右边缘时向内平移；没有可用位置时隐藏而不是半截绘制。`avoidPointLabelCollisions()`（ScatterGraph，含碰撞避让）和 `keepPointLabelsInPlot()`（仅包围盒）为其薄封装。不裁剪的图表不受约束。
- `GPUGraph.tsx` 对比图此前没有任何布局处理，现在在渲染、缩放和标签开关时执行 `keepPointLabelsInPlot`。
- 非官方运行叠加点（`?unofficialrun=`）使用同一套 `.dot-group .point-label` DOM，因此同样受此约束。
- 新增 `point-label-placement.test.ts` 等单元测试；typecheck / lint / fmt / typography 与相关 vitest 套件本地全部通过。

---

> [!NOTE]
> **Low Risk**
> Visualization-only label layout in inference charts; behavior is covered by new jsdom unit tests and does not touch APIs, auth, or data.
> 
> **Overview**
> Fixes scatter/compare chart point labels being **clipped in half** when points sit on the plot edge (e.g. Pareto points at the top of Throughput per Utility MW).
> 
> **Plot bounds:** `plotClipSize()` reads the zoom group’s clip rect in layout coordinates; point-label layout uses that as a **strict** box (flip above/below, slide horizontally at sides, hide when nothing fits).
> 
> **Layout:** Collision logic moves into shared `placePointLabels()` with optional collision avoidance. `avoidPointLabelCollisions()` (single-run scatter) keeps pills/label overlap behavior plus the new bounds; **`GPUGraph`** now runs `keepPointLabelsInPlot()` on render, zoom, and display update so the compare chart gets bounds-only placement.
> 
> Labels hidden only for placement get `data-placement-hidden` so they can reappear after zoom or toggle. Unclipped charts skip the constraint.
> 
> Tests cover edge flips, horizontal shifts, hide/show on zoom, and `plotClipSize` edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 14de2fcadff0d729acac50a34da9f15ca77b0841. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/app/src/components/inference/ui/line-label-layer.ts [hunk 1]
```diff
@@ -432,19 +496,70 @@ export function avoidPointLabelCollisions(
...
       continue;
     }
-    label.firstTspan.setAttribute('dy', `${firstBaselines[index]}px`);
+    const chosen = candidates[index];
+    label.tspans[0].setAttribute('dy', `${chosen.firstY}px`);
+    if (chosen.dx !== 0) {
+      for (const tspan of label.tspans) tspan.setAttribute('x', String(chosen.dx));
+    }
     label.element.style.opacity = '1';
-    placed.push(boxes[index]);
```
[the remark this answers is not in the document]
Comment (cquil11) [reply packages/app/src/components/inference/ui/line-label-layer.ts:448]: Fixed in 14de2fc: labels hidden by the placement pass now carry `data-placement-hidden` and are restored before each layout, so they come back once the point re-enters the plot. Labels hidden by the toggle or legend hover have no marker and are still left alone. Added regression tests for both cases.
